go/common/enclave.go

```diff
@@ -130,3 +132,48 @@ type ProducedSecretResponse struct {
 	RequesterID gethcommon.Address
 	HostAddress string
 }
+
+const (
+	bsePrefix = "blockSubmitError: "
+	l1HeadKey = "l1Head"
+)
+
+var l1HeadErrorSeparator = fmt.Sprintf(", %s=", l1HeadKey)
```

Comment:
I think this is pretty overengineered. I'm having trouble working out what it's doing based on reading the code.

Maybe you just need a const for the separator, then you can either hardcode the equals or use a const for that?

Reply:
I think it was right idea, wrong execution. I think it's important that enclave says, "no, this is where I am" when given a block it's not expecting, multiple RPC calls is clunky.

But this hacky approach at using all the obscure features that Go's error system has to offer was wrong. I've just found the work @tudor-malene did around serializable errors using geth's RPC error types in `evm_facade.go`. `gethrpc.DataError` seems like exactly what we need here.